updateStableIds/src/main/java/org/reactome/release/updateStableIds/UpdateStableIds.java

```diff
@@ -0,0 +1,171 @@
+package org.reactome.release.updateStableIds;
+
+import static org.gk.model.ReactomeJavaConstants.*;
+
+import java.util.*;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.InvalidAttributeException;
+import org.reactome.release.common.database.InstanceEditUtils;
+
+public class UpdateStableIds {
+
+	private static final Logger logger = LogManager.getLogger();
+	
+	@SuppressWarnings("unchecked")
+	public static void stableIdUpdater(MySQLAdaptor dbaSlice, MySQLAdaptor dbaPrevSlice, MySQLAdaptor dbaGkCentral, Long personId) throws Exception {
+		
+		logger.info("Generating InstanceEdits for " + dbaSlice.getDBName() + " and " + dbaGkCentral.getDBName());
+		// Instance Edits for test_slice and gk_central
+		String creatorName = "org.reactome.release.updateStableIds";
+		GKInstance sliceIE = InstanceEditUtils.createInstanceEdit(dbaSlice, personId, creatorName);
+		GKInstance gkCentralIE = InstanceEditUtils.createInstanceEdit(dbaGkCentral, personId, creatorName);
+
+		// At time of writing (December 2018), test_slice is a non-transactional database. This check has been put in place as a safety net in case that changes.
+		if (dbaSlice.supportsTransactions()) {
+			dbaSlice.startTransaction();
+		}
+		dbaGkCentral.startTransaction();
+
+		//TODO: Perl wrapper will create a 'snapshot' of the previous slice -- once the wrapper is retired this needs to be done
```

Comment:
Is this still a "TODO" or is it done?

Reply:
Its still a TODO, since we're still using Perl wrappers for script calling. This is a database step, so for now I've decided to let Perl handle it

Reply:
Ok.